xarray/core/missing.py

```diff
@@ -0,0 +1,261 @@
+
+        self.kind = kind
+        self.cons_kwargs = kwargs
+
+        if fill_value is not None:
+            raise ValueError('from_derivatives does not support fill_value')
+
+        self.f = BPoly.from_derivatives(xi, yi, **self.cons_kwargs)
+
+
+class SplineInterpolator(BaseInterpolator):
+    def __init__(self, xi, yi, kind=None, fill_value=None, order=3, **kwargs):
+        from scipy.interpolate import UnivariateSpline
+
+        if kind is None:
+            raise ValueError('kind is a required argument')
+
+        self.kind = kind
+        self.cons_kwargs = kwargs
+        self.call_kwargs['nu'] = kwargs.pop('nu', 0)
+        self.call_kwargs['ext'] = kwargs.pop('ext', None)
+
+        if fill_value is not None:
+            raise ValueError('SplineInterpolator does not support fill_value')
+
+        self.f = UnivariateSpline(xi, yi, k=order, **self.cons_kwargs)
+
+
+def get_clean_interp_args(arr, method, dim, **kwargs):
+    '''normalize interolation arguments and choose which index to use
+
+    provides compatability with pandas interpolation methods
+    '''
+    axis = arr.get_axis_num(dim)
+    if method == 'linear':
+        index = np.arange(arr.shape[axis])
+    elif method in ['values', 'index']:
+        index = arr.get_index(dim)
+        method = 'linear'
+    elif method == 'time':
+        index = arr.get_index(dim)
+        method = 'linear'
+        assert isinstance(index, pd.DatetimeIndex)
+    elif method == 'polynomial':
+        order = kwargs.get('order', None)
+        if order is None or order < 0:
+            raise ValueError(
+                "order needs to be specified and be >= than 0")
+        index = np.arange(arr.shape[axis])
+    elif method == 'spline':
+        order = kwargs.get('order', None)
+        if order is None or order < 0:
+            raise ValueError(
+                "order needs to be specified and be >= than 0")
+        index = np.arange(arr.shape[axis])
+    else:
+        index = arr.get_index(dim)
+
+    if isinstance(index, pd.DatetimeIndex):
+        index = index.values.astype(np.float64)
+
+    # check index sorting now so we can skip it later
+    if not (np.diff(index) > 0).all():
+        raise ValueError("Index must be monotonicly increasing")
+    return method, index
+
+
+def interp_na(self, dim=None, method='linear', inplace=False, **kwargs):
+    '''Interpolate values according to different methods.'''
+
+    arr = self if inplace else self.copy()
+
+    if dim is None:
+        raise NotImplementedError('dim is a required argument')
+
+    method, index = get_clean_interp_args(arr, method, dim, **kwargs)
+    kwargs.update(kind=method)
+
+    interpolator = _get_interpolator(method, **kwargs)
+
+    return apply_ufunc(wrap_interpolator,
+                       interpolator, index, arr, kwargs=kwargs,
+                       input_core_dims=[[], [dim], [dim]],
+                       output_core_dims=[[dim]],
+                       dask='forbidden',
```

Comment:
I think `dask='parallelized'` would work here -- I tested it with `vectorize=True`.